### ISSUE TITLE
Off-line / air-gapped Panel: implement testing

### ISSUE THREAD

**Issue by Coderambling** [holoviz/panel#7429]
Off-line / air-gapped use cases and related requests pop up regularly.

It would make sense to implement testing for this (using reference apps?), to ensure this works well, and keeps working as new features are added.

As a start, is it posdible to run the current tests, but with internet access disabled? Can this be done using Github Actions?

**Comment by Coderambling** [comment]
Related: https://github.com/holoviz/panel/issues/4257

https://discord.com/channels/1075331058024861767/1113741778496008232/1256570697174745110

General offline implementations: https://offlinefirst.org/

**Comment by Coderambling** [comment]
Related? https://github.com/holoviz/panel/pull/7826

**Comment by Coderambling** [comment]
Related: https://github.com/holoviz/panel/issues/4077

https://github.com/holoviz/panel/issues/7965#issuecomment-2942794662

Discord thread: https://discord.com/channels/1075331058024861767/1390256657497587756

**Comment by mikeymckay** [comment]
You can test offline mode in chrome by opening the developer console, clicking on Network, Throttling and selecting offline. This blocks all outgoing requests for the page. There is a whole community dedicated to offline apps, which are related to progressive web apps (PWAs): https://offlinefirst.org/

**Comment by mikeymckay** [comment]
And note that there may be a workaround/solution found in the comments of: https://github.com/holoviz/panel/issues/4077

**Comment by Coderambling** [comment]
Right. Interesting link!

What I was thinking went I put the sentence below in this issue was:

The Panel library has automated tests on Github, that tests components etc. to catch issues for new releases etc.

So:

1. Copy the tests (or create a new repo for this)?
2. Make it so the tests have access to local resources, but prevent them from going outside the repo to CDN's etc.
3. Look at what the tests report.

So basically simulate offline use of Panel in this way, leveraging the exising tests. That could clarify what is needed to make offline use possible.

I don't know how viable / involved this would be, but that coulld be an approach to identify gaps / issues.


> As a start, is it possible to run the current tests, but with internet access disabled? Can this be done using Github Actions?
